DESIGN.md

```diff
@@ -0,0 +1,60 @@
+# Bulk Data Extract (BDE) Processor design
+
+## User requirements
+
+-  The _user interface_ is a read-only PostgreSQL database in AWS RDS, with
+   Azure authentication
+-  _Accessible_ to all
+   [current BDE users](https://github.com/linz/ds-infra-team/wiki/BDE-users)
+-  Process BDE files
+   _[every day](https://toitutewhenua.atlassian.net/wiki/spaces/GEOD/pages/86418416/BDE+Processor+Product)
+   except Sunday_
+-  Optional _post-processing_ pulled in from separate internal repository
+-  All changes for a single day are imported within a _single transaction_
+-  _Immediate notification_ of completed import (like the legacy BDE Processor
+   SNS topic)
+
+## Technical requirements
+
+-  Thorough testing of every part of the process before merging any pull
+   requests
+-  Separate, managed accounts for infrastructure testing and production
+-  Trivial deployment
+
+## Open questions
+
+-  Can we give read access to the BDE data to the entire organisation? The
+   public?
+
+## Non-requirements
+
+-  Don't keep BDE files around. They should ideally be kept only in memory or at
```

Comment:
We do need to keep an archive of the files. But I like the idea of doing this separately. And then hooking the BDE processor into this archive. I also like the idea of getting rid of the files altogether and using event streaming or something similar to get realtime changes from Landonline.

Reply:
I'll update this, but how long do we need to keep the files? Do we really need them forever, only a couple L0s back, a year, or something else?

Reply:
Getting that info is a good TODO I think.